### PR TITLE
Data tutorial

### PR DESCRIPTION
**Pull request by shaydeci**

### CODE REVIEW

**Review by ofrimasad — COMMENTED** [review]
looks nice.
many small comments

File: documentation/source/Data.md [hunk 1]
```diff
@@ -0,0 +1,263 @@
+# Data in SG
+
+To handle data, SuperGradients takes use of two Pytorch primitives- `torch.utils.data.Dataset` which is in charge of generating the samples and their corresponding labels,
```
[the remark this answers is not in the document]
Comment (shaydeci) [reply documentation/source/Data.md]: fixed in https://github.com/Deci-AI/super-gradients/pull/663/commits/55dbf7a552d260daa9553ca6ec16a36b6feb42ca

File: documentation/source/Data.md [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+    SegmentationDataSet
+    CoCoSegmentationDataSet
+    PascalAUG2012SegmentationDataSet
+    PascalVOC2012SegmentationDataSet
+    CityscapesDataset
+    SuperviselyPersonsDataset
+    PascalVOCAndAUGUnifiedDataset
+    COCOKeypointsDataset
+
+All of which can be imported from the `super_gradients.training.datasets` module. Note that some of the above implementations require following a few simple setup steps, which are all documented [here](https://github.com/Deci-AI/super-gradients/blob/master/src/super_gradients/training/datasets/Dataset_Setup_Instructions.md))
```
[the remark this answers is not in the document]
Comment (shaydeci) [reply documentation/source/Data.md]: fixed in https://github.com/Deci-AI/super-gradients/pull/663/commits/55dbf7a552d260daa9553ca6ec16a36b6feb42ca

File: documentation/source/Data.md [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+Once instantiated, any of the above can be simply passed to the `torch.utils.data.DataLoader` constructor, and be sued for training, validation or testing.
```
[the remark this answers is not in the document]
Comment (shaydeci) [reply documentation/source/Data.md]: https://github.com/Deci-AI/super-gradients/pull/663/commits/962728cffdefcaf6d6294d53d2f010300ee47f51 added

File: documentation/source/Data.md [hunk 4]
```diff
@@ -0,0 +1,263 @@
...
+
+Creating a `torch.utils.data.DataLoader` from some dataset can be tricky, especially when needing to define some parameters on the fly. For example, in distributed training (i.e DDP)
+the  `torch.utils.data.DataLoader` must be given a proper `Sampler`, such that the dataset indices will be divided among the different processes. Not acknowledging this and using the default
+`torch.utils.data.SequentialSampler` will lead the different processes to iterate over the same data samples giving little to no speedup over single GPU training! User beware!
+This is where SG's `training.dataloaders.get` comes in handy, by taking the burden of instantiating the right default sampler according to the training settings.
```
[the remark this answers is not in the document]
Comment (shaydeci) [reply documentation/source/Data.md]: done in https://github.com/Deci-AI/super-gradients/pull/663/commits/a5548897dc32de45f71ed85336795273b6e1064b

File: documentation/source/Data.md [hunk 5]
```diff
@@ -0,0 +1,263 @@
...
+
+dataset = MyDataset(transforms=T.ToTensor())
+dataloader = dataloaders.get(dataset=dataset, dataloader_params={"batch_size": 4})
+```
+
+Note that `dataloader_params` will be unpacked in the `torch.utils.data.DataLoader` constructor, after setting a proper sampler if one is not explicitly set.
+
+## SG DataLoaders
+
+As mentioned above- once instantiated, the `torch.utils.data.DataLoader` objects are the ones forming batches.
```
[the remark this answers is not in the document]
Comment (shaydeci) [reply documentation/source/Data.md]: done https://github.com/Deci-AI/super-gradients/pull/663/commits/6333c61aec683581f15e260b71037f7b24e0c371

File: documentation/source/Data.md [hunk 6]
```diff
@@ -0,0 +1,263 @@
...
+...
+trainer = Trainer("my_imagenet_training_experiment")
+model = ...
+train_params = {...}
+
+trainer.train(model=model, training_params=train_params, train_loader=train_dataloader, valid_loader=valid_dataloader)
+   
+   ```
+
+### SG DataLoaders- Training with Configuration Files
```
[the remark this answers is not in the document]
Comment (shaydeci) [reply documentation/source/Data.md]: done https://github.com/Deci-AI/super-gradients/pull/663/commits/6333c61aec683581f15e260b71037f7b24e0c371

File: documentation/source/Data.md [hunk 7]
```diff
@@ -0,0 +1,263 @@
...
+
+If you are not familiar with training with configuration files, follow see this [link](https://github.com/Deci-AI/super-gradients/tree/master/documentation/source).
```
[the remark this answers is not in the document]
Comment (shaydeci) [reply documentation/source/Data.md]: changed in https://github.com/Deci-AI/super-gradients/pull/663/commits/6333c61aec683581f15e260b71037f7b24e0c371

File: documentation/source/Data.md [hunk 8]
```diff
@@ -0,0 +1,263 @@
...
+
+Any of the SG predefined data loaders listed earlier can be simply plugged in by their names.
```
[the remark this answers is not in the document]
Comment (shaydeci) [reply documentation/source/Data.md]: done in https://github.com/Deci-AI/super-gradients/pull/663/commits/6333c61aec683581f15e260b71037f7b24e0c371

File: documentation/source/Data.md [hunk 9]
```diff
@@ -0,0 +1,263 @@
...
+...
+train_dataloader: imagenet_resnet50_train
+val_dataloader: imagenet_resnet50_val
+
+...
+```
+
+Now, on the structure of `dataset_params`:
+```yaml
+train_dataset_params:
```
[the remark this answers is not in the document]
Comment (shaydeci) [reply documentation/source/Data.md:168]: Added https://github.com/Deci-AI/super-gradients/pull/663/commits/6333c61aec683581f15e260b71037f7b24e0c371

File: documentation/source/Data.md [hunk 10]
```diff
@@ -0,0 +1,263 @@
...
+train_dataloader_params:
+val_dataset_params:
+val_dataloader_params:
+
+```
+As their names suggest- these entries `train_dataset_params` will be passed to the underlying `torch.utils.data.Dataset`
```
[the remark this answers is not in the document]
Comment (shaydeci) [reply documentation/source/Data.md]: added https://github.com/Deci-AI/super-gradients/pull/663/commits/6333c61aec683581f15e260b71037f7b24e0c371

File: documentation/source/Data.md [hunk 11]
```diff
@@ -0,0 +1,263 @@
...
+class constructor and `train_dataloader_params` will be passed to the `torch.utils.data.DataLoader` constructor.
+As in the previous sub-section, both `train_dataloader_params` and `train_dataset_params` will override the the corresponding
+parameters defined for the predefined data loader ( in our case, imagenet_resnet50 recipe's dataset_params.train_dataset_params, and
+imagenet_renet50 recipe's dataset_params.train_dataloader_params).
+This occurs equivalently for the validation set as well.
```
[the remark this answers is not in the document]
Comment (shaydeci) [reply documentation/source/Data.md]: done https://github.com/Deci-AI/super-gradients/pull/663/commits/6333c61aec683581f15e260b71037f7b24e0c371

File: documentation/source/Data.md [hunk 12]
```diff
@@ -0,0 +1,263 @@
...
+To demonstrate, to train with the same data settings as in the previous code snippet:
```
[the remark this answers is not in the document]
Comment (shaydeci) [reply documentation/source/Data.md]: fixed

**Review by ofrimasad — COMMENTED** [review]
Just one sentence that requires rephrasing

File: documentation/source/Data.md [hunk 13]
```diff
@@ -0,0 +1,285 @@
...
+    pascal_aug_segmentation_train
+    pascal_aug_segmentation_val
+    pascal_voc_segmentation_train
+    pascal_voc_segmentation_val
+    supervisely_persons_train
+    supervisely_persons_val
+    pascal_voc_detection_train
+    pascal_voc_detection_val
+
+All of which can be imported from the `super_gradients.training.dataloaders` module, and
```
[the remark this answers is not in the document]
Comment (shaydeci) [reply documentation/source/Data.md]: done in the latest commit.

**Review by ofrimasad — APPROVED** [review]
LGTM
